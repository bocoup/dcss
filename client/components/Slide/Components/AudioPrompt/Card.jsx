import React from 'react';
import { Icon } from '@components/UI';
import { name } from './meta';

const Card = () => (
  <React.Fragment>
    <Icon name="microphone" aria-label={name} />
    {name}
  </React.Fragment>
);

export default React.memo(Card);
