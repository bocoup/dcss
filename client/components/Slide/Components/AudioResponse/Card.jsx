import React from 'react';
import { Popup, Icon } from 'semantic-ui-react';
import { name } from './meta';

const AudioResponseCard = () => (
  <Popup
    content="An audio recorder for users to record responses."
    header={name}
    trigger={
      <React.Fragment>
        <Icon name="microphone" aria-label={name} />
        {name}
      </React.Fragment>
    }
  />
);

export default React.memo(AudioResponseCard);
