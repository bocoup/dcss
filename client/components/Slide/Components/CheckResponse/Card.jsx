import React from 'react';
import { Popup, Icon } from 'semantic-ui-react';

const TextResponseCard = () => (
    <Popup
        content="A checkbox response"
        header="Checkbox Response"
        trigger={
            <React.Fragment>
                <Icon
                    name="check square outline"
                    aria-label="Checkbox Response"
                />
                Checkbox Response
            </React.Fragment>
        }
    />
);

export default React.memo(TextResponseCard);
