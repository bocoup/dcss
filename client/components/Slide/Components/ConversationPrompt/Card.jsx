import React from 'react';
import { Icon } from '@components/UI';
import { name } from './meta';

const Card = () => (
  <React.Fragment>
    <Icon name="conversation" aria-label={name} />
    {name}
  </React.Fragment>
);

export default React.memo(Card);
