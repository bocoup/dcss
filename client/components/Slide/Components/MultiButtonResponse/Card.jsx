import React from 'react';
import { Popup, Icon } from 'semantic-ui-react';
import { name } from './meta';

const MultiButtonResponseCard = () => (
  <Popup
    content="A multiple button response"
    header={name}
    trigger={
      <React.Fragment>
        <Icon name="hand pointer outline" aria-label={name} />
        {name}
      </React.Fragment>
    }
  />
);

export default React.memo(MultiButtonResponseCard);
