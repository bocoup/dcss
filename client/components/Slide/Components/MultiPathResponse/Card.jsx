import React from 'react';
import { Icon } from '@components/UI';
import { name } from './meta';

const Card = () => (
  <React.Fragment>
    <Icon name="fork" rotated="clockwise" aria-label={name} />
    {name}
  </React.Fragment>
);

export default React.memo(Card);
