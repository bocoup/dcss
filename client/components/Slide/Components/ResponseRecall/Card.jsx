import React from 'react';
import { Popup, Icon } from 'semantic-ui-react';
import { name } from './meta';

const ResponseRecallCard = () => (
  <Popup
    content="Embed a previously entered participant response"
    header={name}
    trigger={
      <React.Fragment>
        <Icon name="comment alternate outline" aria-label={name} />
        {name}
      </React.Fragment>
    }
  />
);

export default React.memo(ResponseRecallCard);
