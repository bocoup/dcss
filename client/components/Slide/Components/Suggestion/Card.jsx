import React from 'react';
import { Icon } from '@components/UI';
import { name } from './meta';

const Card = () => (
  <React.Fragment>
    <Icon name="info" aria-label={name} />
    {name}
  </React.Fragment>
);

export default React.memo(Card);
