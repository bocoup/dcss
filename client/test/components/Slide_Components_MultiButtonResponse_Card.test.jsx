import React from 'react';
import assert from 'assert';
import {
  fetchImplementation,
  mounter,
  reduxer,
  snapshotter,
  state,
} from '../bootstrap';
import { unmountComponentAtNode } from 'react-dom';
import { mount, render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Card from '../../components/Slide/Components/MultiButtonResponse/Card.jsx';

const original = JSON.parse(JSON.stringify(state));
let container = null;
let commonProps = null;
let commonState = null;

beforeAll(() => {
  (window || global).fetch = jest.fn();
});

afterAll(() => {
  fetch.mockRestore();
});

beforeEach(() => {
  container = document.createElement('div');
  container.setAttribute('id', 'root');
  document.body.appendChild(container);

  fetchImplementation(fetch);

  commonProps = {
    history: {
      push() {},
    },
  };

  commonState = JSON.parse(JSON.stringify(original));
});

afterEach(() => {
  fetch.mockReset();
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  commonProps = null;
  commonState = null;
});

test('Card', () => {
  expect(Card).toBeDefined();
});

test('Snapshot 1 1', async () => {
  const Component = Card;

  const props = {
    ...commonProps,
  };

  const state = {
    ...{},
  };

  const reduxed = reduxer(Component, props, state);
  const mounted = mounter(reduxed, container);
  expect(snapshotter(mounted)).toMatchSnapshot();
});

test('Snapshot 1 2', async () => {
  const Component = Card;

  const props = {
    ...commonProps,
  };

  const state = {
    ...commonState,
  };

  const reduxed = reduxer(Component, props, state);
  const mounted = mounter(reduxed, container);
  expect(snapshotter(mounted)).toMatchSnapshot();
});

/*{INJECTION}*/

