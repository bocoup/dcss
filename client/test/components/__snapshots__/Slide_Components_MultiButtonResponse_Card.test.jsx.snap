// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render 1 1 1`] = `
<DocumentFragment>
  <i
    aria-label="Prompt: Multiple Buttons"
    class="hand pointer outline icon"
  />
  Prompt: Multiple Buttons
</DocumentFragment>
`;
